lib/class-wp-theme-json-gutenberg.php

```diff
@@ -233,6 +234,7 @@ class WP_Theme_JSON_Gutenberg {
 		'padding-right'              => array( 'spacing', 'padding', 'right' ),
 		'padding-bottom'             => array( 'spacing', 'padding', 'bottom' ),
 		'padding-left'               => array( 'spacing', 'padding', 'left' ),
+		'--wp--style--block-gap'     => array( 'spacing', 'blockGap' ),
```

Comment:
The gap works a little bit differently from the other properties because there is inheritance. If I apply margin to a block the descendent blocks don't get that margin but if I apply a gap the descendants even if not direct descendants get that gap because of how CSS variables work. I guess we should probably document somewhere that gap is inherited/propagated contrary to what happens with a margin or padding.

Reply:
Good observation, I went back and forth here. I think inheriting is good here. WDYT? Agreed about the documentation.

Reply:
I think for this case inheritance makes sense, otherwise, something like a global gap would have almost no noticeable effect. As the direct descendants in a global context are header a content area and a footer most of the time.
I guess we can document and set that expectation for the gap and see what feedback we get.

Reply:
unless I'm missing something, I didn't find a reference for all the styles that we can add the theme.json configs. Maybe we should consider creating one separately.

Reply:
Perhaps within the "styles" section? https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/#styles

Reply:
Yep, I think that would be a good place for it.